src/Matrix/Matrix.h

```diff
@@ -206,15 +206,17 @@ class Matrix : public Matrix_base<T,N> ,public std::enable_shared_from_this<Matr
   Enable_if<Matrix_type<M>(), Matrix<T,N>&> apply(M& m, F f)
   {
     assert(Matrix_impl::same_extents(desc, m.descriptor()));
-    for(auto i = begin(), j = m.begin(); i!=end(); ++i, ++j)
+    auto i = begin();
+    auto j = m.begin();
+    for(; i!=end(); ++i, ++j)
       f(*i,*j);
     return *this;
   }
 
   template<typename M>
   Enable_if<Matrix_type<M>(),Matrix<T,N>&> operator+=(const M& m)
   {
-    static_assert(m.order==N, "+=: mimatched Matrix dimensions");
+    static_assert(M::order==N, "+=: mismatched Matrix dimensions");
```

Comment:
The point is here.
`m.order` is not regarded as compile-time constant by clang.